Dart/src/com/jetbrains/lang/dart/ide/toolingDaemon/DartToolingDaemonServiceConsumer.kt

```diff
@@ -0,0 +1,8 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package com.jetbrains.lang.dart.ide.toolingDaemon
+
+import com.google.gson.JsonObject
+
+fun interface DartToolingDaemonServiceConsumer {
+  fun received(request: JsonObject): JsonObject
```

Comment:
Maybe rename `received` to something like `handleRequest`?

The implementations can't fail and always return valid not null `JsonObject`, right?

Reply:
1. `handleRequest` sounds better! change to `RequestHandler` class name also fine
2. Yes, if there is an error the resulting `JsonObject` should reflect that.

Reply:
done